source/code/go/src/plugins/out_oms.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"time"
+
+	"github.com/fluent/fluent-bit-go/output"
+)
+import (
+	"C"
+	"unsafe"
+)
+
+//export FLBPluginRegister
+func FLBPluginRegister(ctx unsafe.Pointer) int {
+	return output.FLBPluginRegister(ctx, "oms", "Stdout GO!")
+}
+
+//export FLBPluginInit
+// (fluentbit will call this)
+// ctx (context) pointer to fluentbit context (state/ c code)
+func FLBPluginInit(ctx unsafe.Pointer) int {
+	readConfig()
+	go initMaps()
+	go updateIgnoreContainerIds()
+	return output.FLB_OK
+}
+
+//export FLBPluginFlush
+func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
+	var count int
+	var ret int
+	var record map[interface{}]interface{}
+	var records []map[interface{}]interface{}
+
+	start := time.Now()
+	// Create Fluent Bit decoder
+	dec := output.NewDecoder(data, int(length))
+
+	// Iterate Records
+	count = 0
+	for {
+		// Extract Record
+		ret, _, record = output.GetRecord(dec)
+		if ret != 0 {
+			break
+		}
+		records = append(records, record)
+		count++
+	}
+	PostDataHelper(records)
+
+	elapsed := time.Since(start)
+
+	Log("Successfully flushed %d records in %s", len(records), elapsed)
+
+	return output.FLB_OK

```

Comment:
If ODS end point is down, where do we return FLB_RETRY so the engine can try to flush the records at a later time ?